app/src/androidTest/java/com/github/geohunt/app/data/repository/TeamsRepositoryTest.kt

```diff
@@ -0,0 +1,125 @@
+package com.github.geohunt.app.data.repository
+
+import com.github.geohunt.app.data.repository.bounties.TeamsRepository
+import com.github.geohunt.app.mocks.MockUserRepository
+import com.github.geohunt.app.model.database.FirebaseEmulator
+import com.google.firebase.database.FirebaseDatabase
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class TeamsRepositoryTest {
+    private lateinit var database: FirebaseDatabase
+
+    @Before
+    fun setupEmulator() {
+        database = FirebaseEmulator.getEmulatedFirebase()
+    }
+
+    @Test
+    fun createsProperlyATeam() = runTest {
+        val repo = TeamsRepository(
+            bountyReference = database.getReference("bounty/test-bounty"),
+            MockUserRepository()
+        )
+        val t = repo.createTeam("name", "1")
+        assert(
+            getFromDb(database, "bounty/test-bounty/teams/${t.teamId}/teamLeader", String::class.java)
+            == "1"
+        )
+        assert(
+            getFromDb(database, "bounty/test-bounty/teams/${t.teamId}/members/1", Boolean::class.java)
+        )
+        repo.leaveTeam()
+    }
+
+
+    @Test
+    fun joiningATeamEmitsOnSpecificTeamFlow() = runTest {
+        val repo = TeamsRepository(
+            bountyReference = database.getReference("bounty/test-bounty"),
+            MockUserRepository()
+        )
+        val createdTeam = repo.createTeam("name", "1")
+        val flow = repo.getTeam(createdTeam.teamId)
+        val firstInstance = flow.first()
+        assert(!firstInstance.membersUid.contains("2"))
+        repo.joinTeam(createdTeam.teamId, "2")
+        val secondInstance = flow.first()
+        assert(secondInstance.membersUid.contains("2"))
+        // Clean
+        repo.leaveTeam("2")
+        repo.leaveTeam("1")
+    }
+
+    @Test
+    fun joiningTeamAsLoggedUserJoinsProperly() = runTest {
```

Comment:
I'm not sure I understand what this is testing ?

Reply:
joining Team As Logged User Joins Properly.

Reply:
Upwon ouw discwusswon on Discwoduwu, I do agwee with wou ;) uwu